src/ape/contracts/base.py

```diff
@@ -875,33 +877,17 @@ def from_receipt(cls, receipt: ReceiptAPI, contract_type: ContractType) -> "Cont
             contract_type=contract_type,
             txn_hash=receipt.txn_hash,
         )
-        instance._cached_receipt = receipt
+        instance.creation = ContractCreation.from_receipt(receipt)
         return instance
 
-    @property
-    def receipt(self) -> ReceiptAPI:
+    @cached_property
+    def creation(self) -> ContractCreation:
         """
-        The receipt associated with deploying the contract instance,
-        if it is known and exists.
+        Contract creation details: txn_hash, block, deployer, factory, receipt.
         """
-
-        if self._cached_receipt:
-            return self._cached_receipt
-
-        if self.txn_hash:
-            # Hash is known. Use that to get the receipt.
-            try:
-                receipt = self.chain_manager.get_receipt(self.txn_hash)
-            except (TransactionNotFoundError, ValueError, ChainError):
-                pass
-            else:
-                self._cached_receipt = receipt
-                return receipt
-
-        # Brute force find the receipt.
-        receipt = self.chain_manager.contracts.get_creation_receipt(self.address)
-        self._cached_receipt = receipt
-        return receipt
+        data = self.chain_manager.contracts.get_creation_metadata(self.address)
+        assert data, "unreachable"
```

Comment:
Hitting this in tests. Thus it is reachable!

Reply:
does this work on EthTester?

Reply:
i was under impression that `ContractInstance` is guaranteed to point to a live contract. but apparently it's not enforced. in this case we can just propagate None from the query. fixed this in https://github.com/ApeWorX/ape/pull/2001/commits/639c511f457bea5d002585c1aabe3b282f975fcc